cirq/linalg/transformations.py

```diff
@@ -304,3 +317,152 @@ def partial_trace(tensor: np.ndarray,
     left_indices = [keep_map[i] if i in keep_set else i for i in range(ndim)]
     right_indices = [ndim + i if i in keep_set else i for i in left_indices]
     return np.einsum(tensor, left_indices + right_indices)
+
+
+def wavefunction_partial_trace(
```

Comment:
This name is much better than it used to be, but it does suggest that the output is going to be a density matrix.

Maybe `wavefunction_partial_trace_as_mixture`? So long...
Maybe `subwavefunction_mixture`?

Reply:
I like how explicit `wavefunction_partial_trace_as_mixture` is